crates/bevy_ecs/src/query/state.rs

```diff
@@ -751,15 +754,27 @@ impl<D: QueryData, F: QueryFilter> QueryState<D, F> {
         let mut fetch_state = NewD::get_state(world.components()).expect("Could not create fetch_state, Please initialize all referenced components before transmuting.");
         let filter_state = NewF::get_state(world.components()).expect("Could not create filter_state, Please initialize all referenced components before transmuting.");
 
-        NewD::set_access(&mut fetch_state, &self.component_access);
+        let mut self_access;
```

Comment:
Unnecessary line?

Reply:
That's the local variable that holds the `clone()` if we need to make one.  It's only *initialized* in the `if`, but it needs to be *declared* outside the `if` so that it lives long enough to be borrowed from.  And Rust will handle conditionally dropping it!

Reply:
I think we can do this with a little weirdness. I drafted it up in the playground: https://play.rust-lang.org/?version=stable&mode=debug&edition=2024&gist=369649ab4f7a305b17055a0fae83fa5a

I personally think this is preferable since it means we get rid of this rather weird interaction. Generally you need to initialize the variable in both branches - not doing so makes me deeply uncomfortable haha. Also the shadowing seems like a mistake. At least with the internal block it's clear that it's "localized".

Reply:
Oh, using lifetime extension like that is really clever!  The `&{` part is a little odd, but I can extract the body into a local function to avoid that.  I'll make that change!

Reply:
Ah even better!